pygments/lexers/sql.py

```diff
@@ -595,63 +606,149 @@ class MySqlLexer(RegexLexer):
     tokens = {
         'root': [
             (r'\s+', Text),
-            (r'(#|--\s+).*\n?', Comment.Single),
-            (r'/\*', Comment.Multiline, 'multiline-comments'),
+
+            # Comments
+            (r'(?:#|--\s+).*(?=\n)?', Comment.Single),
```

Comment:
Does the optional lookahead assertion do anything?

Reply:
It might not, but during testing it seemed to be the only thing that worked.

The original lexer code consumed a newline if it existed, and I didn't want the newline tokenized as a comment. I tried using `$` to indicate end-of-line but the `#` comment syntax started rendering as an error token. I found that the lookahead gave me the behavior I wanted without introducing errors, but there may still be a better solution for this (or perhaps I made a mistake during testing).

I'm open to suggestions!

Reply:
I've got a nagging feeling that the lookahead isn't necessary, Let me double-check this later today.

Reply:
Yep, this was unnecessary. I removed the lookahead and expanded the single line comment tests as well.